test/com/facebook/buck/android/DxStepTest.java

```diff
@@ -200,6 +201,33 @@ public void testVerbose10AddsVerboseFlagToDx() throws IOException {
     }
   }
 
+  @Test
+  public void testOverridenMaxHeapSize() throws IOException {
+    try (ExecutionContext context = createExecutionContext(2)) {
+      ProjectFilesystem filesystem = FakeProjectFilesystem.createJavaOnlyFilesystem();
+
+      DxStep dx = new DxStep(
+          filesystem,
+          SAMPLE_OUTPUT_PATH,
+          SAMPLE_FILES_TO_DEX,
+          EnumSet.noneOf(DxStep.Option.class),
+          Optional.of("2g"));
+
+      String expected = String.format(
+          "%s --output %s %s",
+          "/usr/bin/dx -JXmx2g --dex",
```

Comment:
FYI, this broke the windows test because it's using hard-coded "/" instead of EXPECTED_DX_PREFIX.  I'm fixing it locally.